lib/mix/lib/mix/tasks/test.ex

```diff
@@ -228,10 +228,33 @@ defmodule Mix.Tasks.Test do
       `["test"]` if the `test` directory exists, otherwise, it defaults to `[]`.
       It is expected that all test paths contain a `test_helper.exs` file
 
-    * `:test_pattern` - a pattern to load test files. Defaults to `*_test.exs`
+    * `:test_pattern` - a pattern to find potential test files.
+      Defaults to `*.{ex,exs}`.
 
-    * `:warn_test_pattern` - a pattern to match potentially misnamed test files
-      and display a warning. Defaults to `*_test.ex`
+      In Elixir versions earlier than 1.19.0, this option defaulted to `*_test.exs`,
+      but to allow better warnings for misnamed test files, it since matches any
+      Elixir file and expects those to be filtered by `:test_load_filters` and
+      `:test_warn_filters`.
+
+    * `:test_load_filters` - a list of files, regular expressions or one-arity
+      functions to restrict which files matched by the `:test_pattern` are loaded.
+      Defaults to `[~r/.*_test\\.exs$/]`
+
+    * `:test_warn_filters` - a list of files, regular expressions or one-arity
```

Comment:
Wait, so this is the opposite, no? Those are files that should be ignored, since we warn on everything else. So perhaps `:test_ignore_filters`? Sorry for all of the back and forth here.

Reply:
Btw, the reason why there was/is confusion is because I thought this would check for the explicit things we want to warn on. I dropped some comments below but in a nutshell there is an issue with the filters here, which will fail for some patterns and perhaps some OSes (namely Windows).

Reply:
You're right, the word filters is used in the opposite way for the load vs warn case, oops. On top, I previously did it the other way round, but I think it is easier to reason about which files NOT to warn about than which files to warn about, especially when it can be a list. Because if we have multiple filters in the list and it says which files should warn, should only one filter need to match, or should it be all of them?

I still think doing it the ignore way is right, so we should probably rename it. I'll look into the Windows path issues.